compiler-builtins/src/int/udiv.rs

```diff
@@ -43,6 +43,144 @@ intrinsics! {
 
         ((rem as u64) << 32) | (div as u64)
     }
+
+    /// Returns unsigned 8-bit `n / d` and `n % d`.
+    ///
+    /// Note: GCC implements a [non-standard calling convention](https://gcc.gnu.org/wiki/avr-gcc#Exceptions_to_the_Calling_Convention) for this function.
+    #[naked]
+    pub unsafe extern "C" fn __udivmodqi4() {
+        // Derived from: https://github.com/gcc-mirror/gcc/blob/95f10974a9190e345776604480a2df0191104308/libgcc/config/avr/lib1funcs.S#L1365
+        
+        // r25: remainder
+        // r24: dividend, quotient
+        // r22: divisor
+        // r23: loop counter
+        core::arch::naked_asm!(
+            "clr r25",      // clear remainder
+            "ldi r23, 8",   // init loop counter
+            "lsl r24",      // shift dividend
+            "rol r25",      // shift dividend into remainder
+            "cp r25, r22",  // compare remainder & divisor
+            "brcs 2",       // remainder <= divisor
```

Comment:
I had to use hard-coded offsets because I couldn't figure out the local label syntax.  Trying to use purely numeric labels I would see the codegen trying to jump to the same label name from a different function which implies I hadn't figured out how to make them local.

Reply:
Mind posting an example of what you tried? Maybe the AVR assembler doesn't handle numeric labels.

@Amanieu may know better.